alarms/alarms.go

```diff
@@ -0,0 +1,106 @@
+// Copyright (c) Abstract Machines
+// SPDX-License-Identifier: Apache-2.0
+
+package alarms
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"github.com/absmach/supermq/pkg/authn"
+)
+
+const (
+	SeverityMax uint8 = 100
+)
+
+var ErrInvalidSeverity = errors.New("invalid severity. Must be between 0 and 100")
+
+type Metadata map[string]interface{}
+
+// Alarm represents an alarm instance.
+type Alarm struct {
+	ID          string    `json:"id"`
+	RuleID      string    `json:"rule_id"`
+	Status      Status    `json:"status"`
```

Comment:
Can we add Channel ID ,Thing ID, Subtopic to this struct?

Reply:
Should it contain them?

The foreign key is a rule which can contain this information

Reply:
The rule can change, so it does make sense to have that info there.